syntax = "proto3";

package atomic;

message AtomicSwapProto {
  string uuid = 1;
  string senderAddress = 2;
  string receiverAddress = 3;
  uint64 amount = 4;
  string secretHash = 5;
  string secret = 6;
  int64 createdAt = 7;
  int64 expiresAt = 8;
  int32 state = 9;

  bytes zkProof = 10;

  bytes falconSignature = 11;       
  bytes dilithiumSignature = 12;    
}
